pkg/controller/manager/manager_controller.go

```diff
@@ -271,19 +272,23 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 		}
 	}
 
-	// Check that compliance is running.
-	compliance, err := compliance.GetCompliance(ctx, r.client)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			r.status.SetDegraded("Compliance not found", err.Error())
+	var installCompliance = utils.IsFeatureActive(license, common.ComplianceFeature)
+
+	if installCompliance {
```

Comment:
There are two if installCompliance blocks, you could add all logic into one.

Reply:
Alternatively, we have this block with a helper method, which could be copied for compliance.

```
	if !utils.IsAPIServerReady(r.client, reqLogger) {
		r.status.SetDegraded("Waiting for Tigera API server to be ready", "")
		return reconcile.Result{}, nil
	}
```

Reply:
Sure, I can combine them. I might be mistaken, but I do not think the sample you provided applies to this PR ...

Reply:
```
if installCompliance && !utils.isComplianceReady (r.client, reqLogger) {
		r.status.SetDegraded("Waiting for Compliance to be ready", "")
		return reconcile.Result{}, nil
	
}
```

Reply:
I would rather not create a function that required a logger as a parameter and keep the logic in this controller.